tools/mason/MasonLogger.chpl

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020-2025 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module MasonLogger {
+  import IO;
+  import this.SafeCalls as Safe;
+
+
+  // Note: we use >= to determine whether we should output something. So, if you
+  // change enum values, make sure to put things in order, where the lower
+  // values mean lower verbosity.
+  enum logLevel { no, error, warn, info, debug };
+
+  config const logs = logLevel.info;
+
+  private var logWriter = IO.stdout;
+
+  private proc doDebug do return logs>=logLevel.debug;
+  private proc doInfo do return logs>=logLevel.info;
+  private proc doWarn do return logs>=logLevel.warn;
+  private proc doError do return logs>=logLevel.error;
+
+  private var pad = 0;
+
+  record logger {
+    var prefix: string;
+
+    proc init(prefix) {
+      this.prefix = prefix;
+
+      pad = max(pad, prefix.size);
+    }
+
+    // TODO make all variadic
+    proc info(s: string) {
+      if doInfo then Safe.writef(logWriter, addPrefix("%s"), s); ;
+    }
+
+    proc infoln(s: string) {
+      if doInfo then Safe.writeln(logWriter, addPrefix(s));
+    }
+
+    proc infof(f: string, args...) {
+      if doInfo then Safe.writef(logWriter, addPrefix(f), (...args));
+    }
+
+    proc warn(s: string) {
+      if doWarn then Safe.writef(logWriter, addPrefix("%s"), s);
+    }
```

Comment:
warn should have a warnln and warnf variant

Reply:
Yeah, I considered making things complete but untested, vs incomplete. I can add them, but I don't want to bend over backwards to use them :D

Reply:
Yeah no need to bend over backwards for that. Maybe add a todo?

Reply:
Based on your previous suggestion for using log.warn, I added them both, tested warnln anecdotally, used warnf.